core/src/main/java/org/apache/iceberg/rest/RESTSessionCatalog.java

```diff
@@ -1239,7 +1239,9 @@ public List<TableIdentifier> listViews(SessionContext context, Namespace namespa
 
   @Override
   public boolean viewExists(SessionContext context, TableIdentifier identifier) {
-    Endpoint.check(endpoints, Endpoint.V1_VIEW_EXISTS);
+    if (!endpoints.contains(Endpoint.V1_VIEW_EXISTS)) {
+      return false;
```

Comment:
Wasn't the previous behavior to attempt to load the view if the HEAD request is not supported? I think it would be safer to check `V1_LOAD_VIEW` and use that if it is supported before returning false.

Reply:
We have some inconsistency about how we're handling this.  The `listViews` takes the same approach as what I have here, which also aligns with our `DEFAULT_ENDPOINTS`, so it's strange that it would call load view, but list view would return nothing.

Reply:
+1 to fallback to older logic as @nastra suggested. I sent #12294 with some tests. 

I decided to skip Iceberg 1.8.0 in Tirno due to this endpoint's issue and S3 strong integrity checksum issue #12264.